components/src/lib/Modal.tsx

```diff
@@ -0,0 +1,76 @@
+import classnames from "classnames";
+import useModalContext, { ModalContext } from "./hooks/useModalContext";
+import useKeyDown from "./hooks/useKeyDown";
+import {
+  ModalButtonProps,
+  ModalCloseButtonProps,
+  ModalContentProps,
+  ModalDimmerProps,
+  ModalMainProps,
+} from "./types/modalProps";
+import style from "./Modal.module.css";
+import { ReactComponent as CloseIcon } from "./assets/closeIcon.svg";
+
+interface IModal extends React.FC<ModalMainProps> {
+  Dimmer: React.FC<ModalDimmerProps>;
+  Content: React.FC<ModalContentProps>;
+  CloseButton: React.FC<ModalCloseButtonProps>;
+  Button: React.FC<ModalButtonProps>;
+}
+
+const Modal: IModal = Object.assign(ModalMain, {
+  Dimmer: ModalDimmer,
+  Content: ModalContent,
+  CloseButton: ModalCloseButton,
+  Button: ModalButton,
+});
+
+export default Modal;
+
+function ModalMain({ children, isOpen, onClose }: ModalMainProps) {
+  useKeyDown("Escape", onClose);
+
+  return (
+    <ModalContext.Provider value={{ onClose }}>
+      <div className={classnames({ [style.ModalContainer]: true, [style.isOpen]: isOpen })}>
```

Comment:
사소한데 항상 true인 것은 바로 인자로 전달하면 더 간결할 것 같아용

```suggestion
      <div className={classnames(style.ModalContainer, { [style.isOpen]: isOpen })>
```